src/components/list/EntryList.js

```diff
@@ -11,8 +12,30 @@ import ListEntry from '../ui/ListEntry'
 import { TypeName } from '../ui/TypeName'
 import { ReactComponent as LeafIcon } from './leaf.svg'
 
-// TODO: use settings for this
-const convertMetersToMiles = (meters) => (meters * 0.000621371192).toFixed(2)
+const METERS_TO_FEET = 0.3048
+const FEET_TO_MILES = 5280
+const CONVERSION_THRESHOLD = 1000
+const METERS_TO_KILOMETERS = 1000
+
+const convertDistance = (distance, setting) => {
+  if (setting === 'imperial') {
+    const feet = Math.round(distance / METERS_TO_FEET)
+    if (feet < CONVERSION_THRESHOLD) {
+      return `${parseFloat(feet.toPrecision(2))} feet`
+    } else {
+      return `${parseFloat((feet / FEET_TO_MILES).toPrecision(2))} miles`
```

Comment:
Ashank, do you need parseFloats in all of these strings?

Reply:
Tried refactoring this out, into different steps, but looks like it needs to be the last thing called for each value :/